awx/main/models/schedules.py

```diff
@@ -205,10 +205,15 @@ def rrulestr(cls, rrule, fast_forward=True, **kwargs):
                     'A valid TZID must be provided (e.g., America/New_York)'
                 )
 
-        if fast_forward and ('MINUTELY' in rrule or 'HOURLY' in rrule):
+        if (
+            fast_forward and
+            ('MINUTELY' in rrule or 'HOURLY' in rrule) and
+            'COUNT=' not in rrule
+        ):
             try:
                 first_event = x[0]
-                if first_event < now():
+                # If the first event was over a week ago...
+                if (first_event - now()).days > 7:
```

Comment:
Similarly, if the first per-minute/per-hour event *is actually in the past week*, it's not going to be expensive, so we can skip this optimization.